src/shared_modules/content_manager/src/components/CtiDownloader.hpp

```diff
@@ -167,18 +193,22 @@ class CtiDownloader : public AbstractHandler<std::shared_ptr<UpdaterContext>>
             {
                 const std::string exceptionMessage {"Error " + std::to_string(statusCode) + " from server: " + message};
 
-                // If there is an error from the server, throw a different exception.
+                if (statusCode == 429)
+                {
+                    throw cti_server_error {exceptionMessage, CtiErrorType::TOO_MANY_REQUESTS};
+                }
+
                 if (statusCode >= 500 && statusCode <= 599)
                 {
-                    throw cti_server_error {exceptionMessage};
+                    throw cti_server_error {exceptionMessage, CtiErrorType::GENERIC_SERVER_ERROR};
                 }
                 throw std::runtime_error {exceptionMessage};
             }};
 
-        constexpr auto INITIAL_SLEEP_TIME {0};
-        auto sleepTime {INITIAL_SLEEP_TIME};
-        auto retryAttempt {0};
+        unsigned int sleepTime {0};
+        unsigned int retryAttempt;
         auto retry {true};
```

Comment:
The variable "retry" seems useless if after a successful call to method "get" we directly return instead of setting it to false.